ansible_collections/arista/avd/roles/eos_designs/python_modules/mlag/__init__.py

```diff
@@ -0,0 +1,361 @@
+                        "name": "LEAF_PEER_L3",
+                        "trunk_groups": [self._trunk_groups_mlag_l3_name],
+                    }
+                }
+            )
+        vlans.update(
+            {
+                self._mlag_peer_vlan: {
+                    "tenant": "system",
+                    "name": "MLAG_PEER",
+                    "trunk_groups": [self._trunk_groups_mlag_name],
+                }
+            }
+        )
+        return vlans
+
+    @cached_property
+    def vlan_interfaces(self):
+        """
+        Return dict with VLAN Interfaces used for MLAG
+
+        May return both the main MLAG VLAN as well as a dedicated L3 VLAN
+        Can also combine L3 configuration on the main MLAG VLAN
+        """
+
+        # Create Main MLAG VLAN Interface
+        main_vlan_interface_name = f"Vlan{self._mlag_peer_vlan}"
+        main_vlan_interface = {}
+        main_vlan_interface["description"] = "MLAG_PEER"
+        main_vlan_interface["shutdown"] = False
+        ip_address = get(self._hostvars, "switch.mlag_ip", required=True)
+        main_vlan_interface["ip_address"] = f"{ip_address}/31"
+        main_vlan_interface["no_autostate"] = True
+        main_vlan_interface["mtu"] = self._p2p_uplinks_mtu
+        if (struct_cfg := get(self._hostvars, "switch.mlag_peer_vlan_structured_config")) is not None:
+            main_vlan_interface["struct_cfg"] = struct_cfg
+
+        if not self._mlag_l3:
+            return {main_vlan_interface_name: main_vlan_interface}
+
+        # Create L3 data which will go on either a dedicated l3 vlan or the main mlag vlan
+        l3_cfg = {}
+        if self._underlay_routing_protocol == "ospf":
+            l3_cfg["ospf_network_point_to_point"] = True
+            l3_cfg["ospf_area"] = get(self._hostvars, "underlay_ospf_area", required=True)
+
+        elif self._underlay_routing_protocol == "isis":
+            l3_cfg["isis_enable"] = get(self._hostvars, "switch.isis_instance_name", required=True)
+            l3_cfg["isis_metric"] = 50
+            l3_cfg["isis_network_point_to_point"] = True
+
+        if get(self._hostvars, "underlay_multicast") is True:
+            l3_cfg["pim"] = {"ipv4": {"sparse_mode": True}}
+
+        if self._underlay_rfc5549 is True:
+            l3_cfg["ipv6_enable"] = True
+
+        if (struct_cfg := get(self._hostvars, "switch.mlag_peer_l3_vlan_structured_config")) is not None:
+            l3_cfg["struct_cfg"] = struct_cfg
+
+        # Add L3 config if the main interface is also used for L3 peering
+        if self._mlag_peer_l3_vlan is None:
+            main_vlan_interface.update(l3_cfg)
+            # Applying structured config again in the case it is set on both l3vlan and main vlan
+            if (struct_cfg := get(self._hostvars, "switch.mlag_peer_vlan_structured_config")) is not None:
+                main_vlan_interface["struct_cfg"] = struct_cfg
+
+            return {main_vlan_interface_name: main_vlan_interface}
+
```

Comment:
rather than overwriting the `struct_cfg` variable in the walrus assignment, better to call it l3_struct_cfg and this way don't need to run a `get` again
```suggestion
        if (l3_struct_cfg := get(self._hostvars, "switch.mlag_peer_l3_vlan_structured_config")) is not None:
            l3_cfg["struct_cfg"] = l3_struct_cfg

        # Add L3 config if the main interface is also used for L3 peering
        if self._mlag_peer_l3_vlan is None:
            main_vlan_interface.update(l3_cfg)
            # Applying structured config again in the case it is set on both l3vlan and main vlan
            if struct_cfg is not None:
                main_vlan_interface["struct_cfg"] = struct_cfg

            return {main_vlan_interface_name: main_vlan_interface}
```

NOTE: I think this could be refactored even further maybe, avoiding to allocate the l3_struct_cfg to the l3_cfg before the update

Reply:
Fixed